addons/purchase/models/purchase.py

```diff
@@ -156,7 +156,10 @@ def name_get(self):
     @api.model
     def create(self, vals):
         if vals.get('name', 'New') == 'New':
-            vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
+            seq_date = None
+            if 'date_order' in vals:
+                seq_date = fields.Date.to_string(fields.Date.to_date(vals['date_order']))
+            vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order', sequence_date=seq_date) or '/'
```

Comment:
seems like `sequence_date` expectes a Date object, i don't think you need the outer ```fields.Date.to_string``` ?

Reply:
Agreed, i don't understand the double conversion. The `to_date` call should be enough. Same for `sale`.